[IMP] account_cashbox: improve internal transfer payments

cav-adhoc opened the pull request:
1. In `account.payment`, when creating the paired internal transfer payment, set the `cashbox_session_id` to the `destination_cashbox_session_id` of the original payment.

cav-adhoc committed:
[IMP] account_cashbox: improve internal transfer payments with cashbox sessions
1. In `account.payment`, when creating the paired internal transfer payment, set the
    `cashbox_session_id` to the `destination_cashbox_session_id` of the original payment.

roboadhoc commented:
[![Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/account-payment/pull/850.png)](https://mergebot.adhoc.ar/ingadhoc/account-payment/pull/850)

cav-adhoc committed:
[IMP] account_cashbox: add tests for internal transfer cashbox session

cav-adhoc committed:
[FIX] account_cashbox_session: restrict user selection to allowed users
Before this commit, when creating or editing an account cashbox session,
the user selection field allowed selecting any user in the system. This could
lead to unauthorized users being assigned to cashbox sessions,
affecting the restrictions in _session_id fields of payments.

cav-adhoc committed:
[FIX] account_cashbox: Fix domain of destination cashbox session field

When selecting the destination cashbox session for an internal transfer payment,
the domain should filter cashbox sessions that are linked to the selected
destination journal. This ensures that users can only select valid cashbox sessions
that correspond to the destination journal of the transfer.

rov-adhoc commented:
@roboadhoc bump

rov-adhoc commented:
@roboadhoc r+ rebase-ff

roboadhoc commented:
Merge method set to rebase and fast-forward.
